pkg/nmstate/policy.go

```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
 	return builder.withInterface(newInterface)
 }
 
+// WithVlanInterface adds VLAN interface configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithVlanInterface(baseInterface string, vlanID uint16) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```

Comment:
No need to do this, the validate method does it automatically

Reply:
removed